src/Cake.Core/Polyfill/EnvironmentHelper.cs

```diff
@@ -31,17 +31,35 @@ public static bool Is64BitOperativeSystem()
         public static PlatformFamily GetPlatformFamily()
         {
 #if NETCORE
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            try
```

Comment:
Why was these `try`-`catch` introduced? At what point will we see `PlatformNotSupportedException`?

Reply:
Not all platforms support this kind of probing.  I saw this in another code base that was doing the same thing to try to determine the environment.